src/map/atcommand.cpp

```diff
@@ -8368,31 +8368,34 @@ ACMD_FUNC(whodrops)
 		sprintf(atcmd_output, msg_txt(sd,1285), item_db.create_item_link( id ).c_str(), id->nameid); // Item: '%s' (ID:%u)
 		clif_displaymessage(fd, atcmd_output);
 
-		if (id->mob[0].chance == 0) {
+		if (id->mobs.empty()) {
 			strcpy(atcmd_output, msg_txt(sd,1286)); //  - Item is not dropped by mobs.
 			clif_displaymessage(fd, atcmd_output);
-		} else {
-			sprintf(atcmd_output, msg_txt(sd,1287), MAX_SEARCH); //  - Common mobs with highest drop chance (only max %d are listed):
-			clif_displaymessage(fd, atcmd_output);
+			return 0;
+		}
+		sprintf(atcmd_output, msg_txt(sd,1287), MAX_SEARCH); //  - Common mobs with highest drop chance (only max %d are listed):
+		clif_displaymessage(fd, atcmd_output);
 
-			for (uint16 j=0; j < MAX_SEARCH && id->mob[j].chance > 0; j++)
-			{
-				int dropchance = id->mob[j].chance;
-				std::shared_ptr<s_mob_db> mob = mob_db.find(id->mob[j].id);
-				if(!mob) continue;
+		// NB: We don't const-ref this because we modify dropchance, so let's just make a copy
```

Comment:
Wouldnt it be more performant to const-ref this and just have a separate variable for the "real" droprate calculation?

Reply:
ehh, two 32bit copy or a 64bit address with a 32bit copy, not sure which is better